torch/utils/_pytree.py

```diff
@@ -290,6 +290,18 @@ def register_dataclass(cls: type[Any]) -> None:
     torch.export.register_dataclass(cls)
 
 
+def _constant_flatten(x):  # type: ignore[no-untyped-def]
+    return [], ConstantNode(x)
+
+
+def _constant_unflatten(_, context):  # type: ignore[no-untyped-def]
+    return context.value
+
+
+def _constant_flatten_with_keys(x):  # type: ignore[no-untyped-def]
+    return [], ConstantNode(x)
+
```

Comment:
can you make it clearer that we're going to assume register_constant'ed nodes are immutable in the docs?

Reply:
In which doc?

Reply:
https://github.com/pytorch/pytorch/blob/main/torch/utils/_pytree.py#L294-L307

If we decide that we are requiring a non-trivial `__eq__`, then that's also something to toss into the doc. We also probably need a `__hash__` for caching purposes

Reply:
Ah yeah lemme do that.